app/code/Magento/Backend/App/Action/Context.php

```diff
@@ -105,6 +115,8 @@ public function __construct(
         \Magento\Backend\Model\UrlInterface $backendUrl,
         \Magento\Framework\Data\Form\FormKey\Validator $formKeyValidator,
         \Magento\Framework\Locale\ResolverInterface $localeResolver,
+        \Magento\Framework\Serialize\Serializer\Json $jsonSerializer,
```

Comment:
It's not a good idea to add new dependencies to Context, please inject required dependencies to the class where they really needed.

Reply:
@sidolov thanks for this tip. Is there some reference to me where I could understand when it's a good idea to make improvements `context` classes and when not? I'd like to understand to core code principles better.